src/python_testing/TC_IDM_4_2.py

```diff
@@ -67,6 +67,29 @@ async def get_idle_mode_duration_sec(self, ctrl, ep=ROOT_NODE_ENDPOINT_ID):
             attribute=Clusters.IcdManagement.Attributes.IdleModeDuration
         )
 
+    def get_attribute_value_wait(self, sub, typed_attr_path):
+        start_time = time.time()
+        timeout = 10
+        increment = 0.1
+        loop = True
+        attribute_value = None
+        while loop:
+            # Get the attribute value
+            attribute_value = sub.GetAttribute(typed_attr_path)
+
+            # Check if the value is not an empty string
+            if attribute_value != "":
+                loop = False  # Exit the loop
```

Comment:
```suggestion
                return attribute_value
```
Then you can get rid of the loop var and just loop on time.time() < end_time and throw the error at the end